genai/example_test.go

```diff
@@ -162,32 +183,200 @@ func ExampleGenerativeModel_CountTokens_contextWindow() {
+	fmt.Println("input_token_limit:", info.InputTokenLimit)
+	fmt.Println("output_token_limit:", info.OutputTokenLimit)
 
 }
 
-func ExampleGenerativeModel_CountTokens() {
+func ExampleGenerativeModel_CountTokens_textOnly() {
 	ctx := context.Background()
 	client, err := genai.NewClient(ctx, option.WithAPIKey(os.Getenv("GEMINI_API_KEY")))
 	if err != nil {
 		log.Fatal(err)
 	}
 	defer client.Close()
 
-	model := client.GenerativeModel("gemini-1.5-pro")
-	model.SystemInstruction = &genai.Content{
-		Parts: []genai.Part{genai.Text("You are an expert ichthyologist.")},
+	model := client.GenerativeModel("gemini-1.5-flash")
+	prompt := "The quick brown fox jumps over the lazy dog"
+
+	tokResp, err := model.CountTokens(ctx, genai.Text(prompt))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Println("total_tokens:", tokResp.TotalTokens)
+
+	resp, err := model.GenerateContent(ctx, genai.Text(prompt))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Println("prompt_token_count:", resp.UsageMetadata.PromptTokenCount)
+	fmt.Println("candidates_token_count:", resp.UsageMetadata.CandidatesTokenCount)
+	fmt.Println("total_token_count:", resp.UsageMetadata.TotalTokenCount)
+
+}
+
+func ExampleGenerativeModel_CountTokens_cachedContent() {
+	ctx := context.Background()
+	client, err := genai.NewClient(ctx, option.WithAPIKey(os.Getenv("GEMINI_API_KEY")))
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer client.Close()
+
+	txt := strings.Repeat("George Washington was the first president of the United States. ", 3000)
+	argcc := &genai.CachedContent{
+		Model:    "gemini-1.5-flash-001",
+		Contents: []*genai.Content{{Role: "user", Parts: []genai.Part{genai.Text(txt)}}},
+	}
+	cc, err := client.CreateCachedContent(ctx, argcc)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer client.DeleteCachedContent(ctx, cc.Name)
+
+	modelWithCache := client.GenerativeModelFromCachedContent(cc)
+	prompt := "Summarize this statement"
+	tokResp, err := modelWithCache.CountTokens(ctx, genai.Text(prompt))
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Println("total_tokens:", tokResp.TotalTokens)
+
+	resp, err := modelWithCache.GenerateContent(ctx, genai.Text(prompt))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Println("prompt_token_count:", resp.UsageMetadata.PromptTokenCount)
+	fmt.Println("candidates_token_count:", resp.UsageMetadata.CandidatesTokenCount)
+	fmt.Println("cached_content_token_count:", resp.UsageMetadata.CachedContentTokenCount)
+	fmt.Println("total_token_count:", resp.UsageMetadata.TotalTokenCount)
+
+}
+
+func ExampleGenerativeModel_CountTokens_imageInline() {
+	ctx := context.Background()
+	client, err := genai.NewClient(ctx, option.WithAPIKey(os.Getenv("GEMINI_API_KEY")))
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer client.Close()
+
+	model := client.GenerativeModel("gemini-1.5-flash")
+	prompt := "Tell me about this image"
+	imageFile, err := os.ReadFile(filepath.Join(moduleRootDir(), "genai", "testdata", "personWorkingOnComputer.jpg"))
```

Comment:
Why do we need this? Can't we use a relative directory?
And anyway, there is no `// Output:` comment, so this code won't be run.

Reply:
I add an `// Output` line when I try running the sample locally.

The reason I added it is because it's copied into two distinct directories. Where we develop it `internal/snippets` and the "real" place, where users (or maybe we) want to run these too. So a relative directory wouldn't work.

Hope this makes sense...

Reply:
I don't want people to think this is a recommended practice. Why don't you make a global for this called "testFileDir" or something bland like that, and initialize it to that filepath.Join.

Reply:
Done, and added a comment to clarify. To make this less distracting in the rendered examples I moved the `moduleRootDir` function into an internal package too